src/frequenz/sdk/timeseries/formula_engine/_formula_steps.py

```diff
@@ -417,86 +433,69 @@ def stream_name(self) -> str:
     def _is_value_valid(self, value: QuantityT | None) -> bool:
         return not (value is None or value.isnan() or value.isinf())
 
+    async def _fetch_from_fallback(
+        self, fallback_fetcher: FallbackMetricFetcher[QuantityT]
+    ) -> Sample[QuantityT] | None:
+        try:
+            return await fallback_fetcher.receive()
+        except ReceiverStoppedError:
+            if self._is_stopped:
+                _logger.debug(
+                    "Stream for fallback metric fetcher %s closed.",
+                    fallback_fetcher.name,
+                )
+            else:
+                _logger.error(
+                    "Failed to fetch next value from %s. Fallback stream closed.",
+                    self._name,
+                )
+            return None
+        except ReceiverError[Any] as err:
```

Comment:
This is still wrong, and if for any reason an exception that is not a `ReceiverStoppedError` is raised, we'll get the same error about the exception not inheriting from `BaseException`.

It should be:
```suggestion
        except ReceiverError as err:
```

Reply:
Done